cpp/tests/utilities/column_utilities.cu

```diff
@@ -136,6 +137,19 @@ void expect_equal_buffers(void const* lhs, void const* rhs,
                             typed_rhs));
 }
 
+// copy column bitmask to host (used by to_host())
+std::vector<bitmask_type> bitmask_to_host(cudf::column_view const& c) {
+  std::vector<bitmask_type> host_bitmask;
+  if (c.nullable()) {
+    auto num_bitmasks = bitmask_allocation_size_bytes(c.size()) / sizeof(bitmask_type);
+    host_bitmask.resize(num_bitmasks);
+    CUDA_TRY(cudaMemcpy(host_bitmask.data(), c.null_mask(), num_bitmasks * sizeof(bitmask_type),
+                        cudaMemcpyDeviceToHost));
+  }
+  return host_bitmask;
```

Comment:
Originally, I coded this using an IILE (Immediately Invoked Lambda Expression) to write a more declarative piece of code and avoid having to use `.resize`, but then realized you don't even need the lambda:
```cpp
    if (c.nullable()) {
      auto num_bitmasks = bitmask_allocation_size_bytes(c.size()) / sizeof(bitmask_type);
      std::vector<bitmask_type> = host_bitmask(num_bitmasks);
      CUDA_TRY(cudaMemcpy(host_bitmask.data(), c.null_mask(), num_bitmasks * sizeof(bitmask_type),
                          cudaMemcpyDeviceToHost));
      return host_bitmask;
    } else {
      return std::vector<bitmask_type>{};
    }
```
I try avoid writing code that declares `x`, modifies it based on conditionals and then returns it after conditionals. The more declarative the better.

Reply:
How is your version more *declarative*?

Reply:
The previous code mutates state twice (`.resize()` and `cudaMemcpy`) whereas the revised code only mutates state once (`cudaMemcpy`). In Kotlin, the `if` is not a _statement_ but an _expression_. If we write our code where each branch (`if` and `else`) returns, we are in essence mocking the Kotlin `if` expression. 

Expressions are in general preferable to statements as statements imply mutation. Ben Deane has a great talk on this (one of my all time favorites - I shared it with @jrhemstad) https://youtu.be/2ouxETt75R4?t=339 (hyperlink to specific time point where he talks about expressions vs statements)

Reply:
I missed the resize in the original code (the code I modified). Kotlin schmotlin.

Reply:
BTW, you are reviewing code from a different PR. :) (the bitmask_to_host PR).